src/reducers/vis-state-updaters.js

```diff
@@ -799,10 +830,18 @@ export const removeFilterUpdater = (state, action) => {
     ...state.filters.slice(idx + 1, state.filters.length)
   ];
 
+  const filter = state.filters[idx];
+
   const newState = {
     ...state,
-    datasets: applyFiltersToDatasets(dataId, state.datasets, newFilters),
-    filters: newFilters
+    datasets: applyFiltersToDatasets(dataId, state.datasets, newFilters, state.layers),
+    filters: newFilters,
+    editor: {
+      ...state.editor,
+      features: filter.type !== FILTER_TYPES.polygon ?
+        state.editor.features : state.editor.features.filter(f => f.id !== filter.value.id),
+      selectedFeature: null
```

Comment:
if `filter.type !== FILTER_TYPES.polygon ` should `selectedFeature` be set to null?
Looks like you only need to update editor if ``filter.type === FILTER_TYPES.polygon` so why not
```js

  const newState = {
    ...state,
    datasets: applyFiltersToDatasets(dataId, state.datasets, newFilters, state.layers),
    filters: newFilters,
    editor:  filter.type === FILTER_TYPES.polygon ? {
      ...state.editor,
      features: state.editor.features.filter(f => f.id !== filter.value.id),
      selectedFeature: null
    } : state.editor
  };
```